packages/rest/src/client/AbstractRestClient.ts

```diff
@@ -196,24 +197,13 @@ export abstract class AbstractRestClient {
 
     /**
      * Perform the actual http REST call with appropriate user input
-     * @param {string} resource - URI for this request
-     * @param {string} request - REST request type GET|PUT|POST|DELETE
-     * @param {any[]} reqHeaders - option headers to include with request
-     * @param {any} writeData - data to write on this REST request
-     * @param responseStream - stream for incoming response data from the server. If specified, response data will not be buffered
-     * @param requestStream - stream for outgoing request data to the server
-     * @param normalizeResponseNewLines - streaming only - true if you want newlines to be \r\n on windows
-     *                                    when receiving data from the server to responseStream. Don't set this for binary responses
-     * @param normalizeRequestNewLines -  streaming only - true if you want \r\n to be replaced with \n when sending
-     *                                    data to the server from requestStream. Don't set this for binary requests
-     * @param task - task that will automatically be updated to report progress of upload or download to user
+     * @param {IRestOptions} options
+     * @returns {Promise<string>}
      * @throws  if the request gets a status code outside of the 200 range
      *          or other connection problems occur (e.g. connection refused)
+     * @memberof AbstractRestClient
      */
-    public performRest(resource: string, request: HTTP_VERB, reqHeaders?: any[], writeData?: any,
-                       responseStream?: Writable, requestStream?: Readable,
-                       normalizeResponseNewLines?: boolean, normalizeRequestNewLines?: boolean,
-                       task?: ITaskWithStatus): Promise<string> {
+    public performRestCall(options: IRestOptions): Promise<string> {
```

Comment:
This is pure opinion, but what about a shorter function name for the go-forward REST function?  For example, you had `request` or perhaps `call` or `send` or something else?

Reply:
Yeah, I thought `request` was not specific enough, but I was thinking from a substantive perspective, not a verb perspective 😛 . I will switch it back to `request`.

Reply:
done with a2f090a